### ISSUE TITLE
Upgrade General Availiability APIs before upgrading to Kubernetes v1.26 for GKE

### ISSUE THREAD

**Issue by sgibson91** [2i2c-org/infrastructure#2515]
### Context

I got the below email from GCP and wanted to document it incase it requires action from the wider team. I note that only the Pangeo-Columbia account is mentioned in the email, but thought it best to disseminate widely just in case.

---

We’re writing to let you know that the automatic upgrades from Kubernetes v1.25 to v1.26 will start on or after May 25, 2023. For this reason, we recommend updating all your clusters to use GA API before v1.25 reaches End of Life on February 29, 2024, to ensure continued API functionality.

What do you need to know?
[Kubernetes project’s lifecycle](https://notifications.google.com/g/p/ACUgQrHcxbCnBJPAAQ2FSAJqiaeaeCs8N2nt-P6sxIFccg9ECy30mN8DU-5U8L9m6761OML06XG-EKcbll99WnVZjsFrYM_5FZrfNIMke7-3K0A3rjLJVy1eeWOy9JavvbiS-jDeeDMIq26OLRx2ChukFLaGhNKa1PUJtsR4i4DFaDh1qhGhhOu93l9QGd9184WYPQH1Lc9uB-E-aHyVYuefhLoCRmw6ES3j4q-nPWWIwA) states that once the APIs graduate to general availability, previous alpha and [beta APIs](https://notifications.google.com/g/p/ACUgQrFdVTA9nYpyDLDnIiDx2DnutckXwR3XFFUT4peTRFTpjha4YEOiGLseUxWYvJzTM7efbWAESlvhn0kL11vTUTDMa7QnpfPdmWiIdblwtpLpAHYS2yKGNesTvJM9sQRpHbufS7ufyW_mOFTyGLZmRCtFRWYVzyuydY_5pg85UFE590-cjNfKAj14TZ8N3F8iXDnzI01zQAI3ynnjwsTjW0Vt5fciV7WRaozN-_yvbuf2i3rfYYoZ) are eventually removed. Once beta APIs have been [removed from Kubernetes v1.26](https://notifications.google.com/g/p/ACUgQrHwwaGyFDWUslpJPvP_m5AMTnZcyeUid7eGsIIMLHW4eHHcxIpAY-Evz_xan2Xcw6y2Kw2GG8tyZOKxiudaZRcWLmDDQGgKP0Xxf0_iDFgniFEfkiLcJS8qOZocbqkYM90ZqAv3ayLZMShRLotmC3gusPc4doMpNSPiNYhHFc9VNDdCgD3SbN9o_pxT_crORpGruTupNklvr0meHX4R8j9JXjd-GcaPgPHN-5scL65F48M954k), you can expect to experience some of the following behaviors:

If GKE detects usage of the removed beta APIs in your clusters before upgrading to Kubernetes v1.26, Google will [pause automatic minor upgrades ](https://notifications.google.com/g/p/ACUgQrG45-0kR8FhUfTKkDc_ViycuihihOWFr7BUG33Or2mYCvFIcJl6LdEPdRTyrrvcMSGENsr_YHTWr0a4_5BmsDkQNVXA3gQK8RgFk3-sZp_gQvxo6T_sIdcyF13pqY80uVpAwsN19KLHMcO4lQ-4OcySlb3LJ2TcT5WZG1j4tU-rboCaMbY7Kbo04J4zRGYawcgZ9SGo4efoZu88lgSUiQLoDTLtHzhK4IswRqVEpd0pD2zDf7fECYQs3dewrXE)until Kubernetes v1.25 reaches EoL for GKE clusters on that version. The minor upgrade pause is intended to let you perform a timely and nondisruptive transition from beta APIs removed in Kubernetes v1.26 to their newer supported versions.
API clients won’t be able to use the removed beta APIs in v1.26.
[Manifests](https://notifications.google.com/g/p/ACUgQrEiU9a0w8wu2kts33J5O7qTFc01DRANPOEdFoEfbY3O3IF3X2hhOe3hDFbWgPcCeZ6zcEGmJBZ_gbNrsm_VWXTvNm32fHdkR-vBSMbLUe2T0a6IapGEx_Droh2Wx_WlNGsfAOuZ1fyNvUiV669FxAWtdV4-c3JRCEgHCIUGjr1NFLVr67rJ5KtEAJ4S9pl8kNorqYeXxkZNS1lJ1BcThaQbZbqCVf5OFS-B5XGVaWUA81kzbxKtysI5U-Nu) using the removed beta APIs removed in Kubernetes v1.26 can no longer be applied.
Previously persisted objects will remain functional and can be viewed and updated using the new API versions before and after upgrading to Kubernetes v1.26.
What can you expect after Kubernetes v1.25 reaches EoL?
We will automatically upgrade your clusters remaining on Kubernetes v1.25 to Kubernetes v1.26. Optionally, you can configure [maintenance windows and exclusions](https://notifications.google.com/g/p/ACUgQrEgrtPzY7KZz1lSCRJxyVl14Y3zdZ7gxohEr0nMivcD4APYEef8bqVcEek-7tE8cBUkGSRNl91Xt2QjIqUtFNgbkpIGV-ecGPnlgSheWoI5czabdyWZtqAoSJHJAwnl8oDYYHCcwFoOm83g8ILIY4Kl3r7LuUnMlyt_ip7mPbdCsEUVzCCyGEzXAIIaZiwPEr0LC6Dk28FjILwC2UbwcSa5sscDe9lJWYHUEYq7lonYAq3sPVty78aWxPb7Hr7s4k3g4U5XC6PPyZtG) to control scope and timing of automatic upgrades or to temporarily pause them. Cluster control planes are [upgraded regularly](https://notifications.google.com/g/p/ACUgQrHdO7_3e6b0eg86c9sM32FbrKAn1VCcYX82UWP2d-PujpM6dMXxtAhw5D_bBh99ZSbqKHDmtn34X5dcHUiFpRfzGuwa2GuIf5LffJG3x_HWveiXFd0VU8MlI-b9445jC6Lej9SyAO3cAERM_XDg16EgvYtJ1PLMm7pdSiaCjO5ZnHgKpXf53ChT_0xHwixi8F2kBPmlkY71_Fux3zlauI5YdhxsvvptNnqxeNHi7lWLg_bXpxibbbM).

Please be advised that once Kubernetes v1.25 reaches [EoL](https://notifications.google.com/g/p/ACUgQrG3cYYtfnPA8yD5k8HSeMUqvOEKrukmurk4n33FIkCBevKTvFQ-uMWo_lamvATCNJ0hVEnn_hU9HY6FXRjfadgo9gsAcyadNJrdLcdMYQEoSzOGdl-btFL6Z2IHBP8U9m4i0rWtdgbS_DNEEqMe475PhdlSn8jFB67VyCMJqdqYJJshjEsAWiwq1TOQ2xiPlD1P812H8zUtbOR14lXiIMY6iORfvfmcK-7hwtJJBFpZju7kHJGtq3Bez2gqDu9mhfaki51wjRIb0vlCO_Q) and Google upgrades your clusters to Kubernetes v1.26, you will no longer be able to access the removed beta API versions. Any remaining dependencies on these APIs may stop working.

The GKE [release schedule](https://notifications.google.com/g/p/ACUgQrEwSXO8lUg2tPs0GNUvWPWCMfkK0ZowRmQbJlV5skGdW3aWqG8LPEJ1FIEOmkWbS6X8ntroCY4JgxbX-C8IZTehXXK4CcJMKg5M0vMXJZZmJc6GP_j_FSHDt5eMG9Jrhor3NulWB5JkQaBllDfh0lKE5bOWUXKy-vF1mM4Djz45atpbGxm5VJkS4nN1pk1AigxEQdRKLuh3mqv6MwauueGM0VI9Mr6h21u-QFbbzhE) outlines the cluster upgrade schedule for Kubernetes v1.26. Please note the automatic upgrade start dates for the release channels and for clusters on a static version.

What do you need to do?
We recommend updating your software to use [supported](https://notifications.google.com/g/p/ACUgQrHwwaGyFDWUslpJPvP_m5AMTnZcyeUid7eGsIIMLHW4eHHcxIpAY-Evz_xan2Xcw6y2Kw2GG8tyZOKxiudaZRcWLmDDQGgKP0Xxf0_iDFgniFEfkiLcJS8qOZocbqkYM90ZqAv3ayLZMShRLotmC3gusPc4doMpNSPiNYhHFc9VNDdCgD3SbN9o_pxT_crORpGruTupNklvr0meHX4R8j9JXjd-GcaPgPHN-5scL65F48M954k) GA API versions. Updating your software before upgrading your clusters to Kubernetes v1.26 will ensure your API clients and resource manifests can continue to access and update API resources without disruption.

Note: To assess the use of the removed APIs in your clusters, you can use [Admin Activity log query](https://notifications.google.com/g/p/ACUgQrHwwaGyFDWUslpJPvP_m5AMTnZcyeUid7eGsIIMLHW4eHHcxIpAY-Evz_xan2Xcw6y2Kw2GG8tyZOKxiudaZRcWLmDDQGgKP0Xxf0_iDFgniFEfkiLcJS8qOZocbqkYM90ZqAv3ayLZMShRLotmC3gusPc4doMpNSPiNYhHFc9VNDdCgD3SbN9o_pxT_crORpGruTupNklvr0meHX4R8j9JXjd-GcaPgPHN-5scL65F48M954k) with Google Cloud’s operations suite enabled, to show the Kubernetes v1.26 APIs in use.

Your projects below use the beta APIs removed in v1.26 and will require you to migrate to the newer API versions before your clusters are upgraded to v1.26:

[pangeo-integration-te (pangeo-integration-te-3eea)](https://notifications.google.com/g/p/ACUgQrHnF6m1mRYcmqubzXiePErOPvZYn5C_9Z4BSacM2aejeDkIQ4cbNEi37iaZ0mUfQKofHA0pgfDnhU3U74CTv7sdT-BACVFn4mpYvHdOqjab7itRgE9LMgS1KoFUaFhQoPKbV5-_RYy6-5UVB9zctyNP4ojClBPb-YEOcCrX60iXELH1Gza2G_N3Dbrof7bI1xV8jMMU9ciNzEAP05U0srtyVf5CCna-2wMEe1O-vbCGEGk)
pangeo-hubs-cluster

**Comment by consideRatio** [comment]
I think our clusters are setup to not automatically upgrade by not being in a release channel, but instead being in a fixed version and a "hands off, we'll upgrade this manually" mode. And if we aren't, this is scheduled to happen around March 2024 and not earlier.

I did a quick scan for k8s resource with `apiVersion: v1beta1` etc that may needs to be upgraded to `apiVersion: v1` as k8s versions are released, but I couldn't find anything that I think would cause issues for us if we upgraded to k8s 1.26. I'm also confident our chart dependencies aren't using something incompatible with k8s 1.26.

Still...

>Your projects below use the beta APIs removed in v1.26 and will require you to migrate to the newer API versions before your clusters are upgraded to v1.26:
>
>[pangeo-integration-te (pangeo-integration-te-3eea)](https://notifications.google.com/g/p/ACUgQrHnF6m1mRYcmqubzXiePErOPvZYn5C_9Z4BSacM2aejeDkIQ4cbNEi37iaZ0mUfQKofHA0pgfDnhU3U74CTv7sdT-BACVFn4mpYvHdOqjab7itRgE9LMgS1KoFUaFhQoPKbV5-_RYy6-5UVB9zctyNP4ojClBPb-YEOcCrX60iXELH1Gza2G_N3Dbrof7bI1xV8jMMU9ciNzEAP05U0srtyVf5CCna-2wMEe1O-vbCGEGk)
>pangeo-hubs-cluster

I sadly don't have access to this specific cluster via the GCP cloud console, which makes identifying whats outdated a bit hard. Looking into it I couldn't find something obviously outdated.
